examples/2_Intermediate/stage_two_optimization_finitebuild.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+r"""
+In this example we solve a FOCUS like Stage II coil optimisation problem: the
+goal is to find coils that generate a specific target normal field on a given
+surface.  In this particular case we consider a vacuum field, so the target is
+just zero.
+
+The objective is given by
+
+    J = (1/2) \int |B dot n|^2 ds + alpha * (sum CurveLength) + beta * MininumDistancePenalty
+
+if alpha or beta are increased, the coils are more regular and better
+separated, but the target normal field may not be achieved as well.
+
+The target equilibrium is the QA configuration of arXiv:2108.03711.
+"""
+
+import os
+from pathlib import Path
+import numpy as np
+from scipy.optimize import minimize
+from simsopt.geo.surfacerzfourier import SurfaceRZFourier
+from simsopt.objectives.fluxobjective import SquaredFlux
+from simsopt.geo.curve import curves_to_vtk, create_equally_spaced_curves
+from simsopt.field.biotsavart import BiotSavart
+from simsopt.field.coil import Current, coils_via_symmetries
+from simsopt.geo.curveobjectives import CurveLength, MinimumDistance
+from simsopt.geo.curvefilament import create_multifilament_grid
+
+# Number of unique coil shapes, i.e. the number of coils per half field period:
+# (Since the configuration has nfp = 2, multiply by 4 to get the total number of coils.)
+ncoils = 4
+
+# Major radius for the initial circular coils:
+R0 = 1.0
+
+# Minor radius for the initial circular coils:
+R1 = 0.5
+
+# Number of Fourier modes describing each Cartesian component of each coil:
+order = 5
+
+# Weight on the curve lengths in the objective function:
+ALPHA = 1e-6
+
+# Threshhold for the coil-to-coil distance penalty in the objective function:
+MIN_DIST = 0.1
+
+# Weight on the coil-to-coil distance penalty term in the objective function:
+BETA = 10
+
+# Number of iterations to perform:
+ci = "CI" in os.environ and os.environ['CI'].lower() in ['1', 'true']
+MAXITER = 50 if ci else 400
+
+# File for the desired boundary magnetic surface:
+TEST_DIR = (Path(__file__).parent / ".." / ".." / "tests" / "test_files").resolve()
+filename = TEST_DIR / 'input.LandremanPaul2021_QA'
+
+# Directory for output
+OUT_DIR = "./output/"
+os.makedirs(OUT_DIR, exist_ok=True)
+
+#######################################################
+# End of input parameters.
+#######################################################
+
+# Initialize the boundary magnetic surface:
+nphi = 32
+ntheta = 32
+s = SurfaceRZFourier.from_vmec_input(filename, range="half period", nphi=nphi, ntheta=ntheta)
+
+# Create the initial coils:
+numfilaments_n = 1
+numfilaments_b = 5
+gapsize_n = 0.01
+gapsize_b = 0.01
```

Comment:
How about moving these 4 parameters from lines 74-77 up to around line 52 so they are with the other parameters. Also it would be helpful to explain in comments what `_n` and `_b` and `gapsize` mean.